spec/param_parser_spec.cr

```diff
@@ -2,101 +2,98 @@ require "./spec_helper"
 
 describe "ParamParser" do
   it "parses query params" do
-    route = Route.new "POST", "/" do |env|
-      hasan = env.params.query["hasan"]
-      "Hello #{hasan}"
-    end
     request = HTTP::Request.new("POST", "/?hasan=cemal")
     query_params = Kemal::ParamParser.new(request).query
     query_params["hasan"].should eq "cemal"
   end
 
-  it "parses multiple values for query params" do
-    route = Route.new "POST", "/" do |env|
-      hasan = env.params.query["hasan"]
-      "Hello #{hasan}"
+  context "#query" do
+    it "parses multiple values for query params" do
+      request = HTTP::Request.new("POST", "/?hasan=cemal&hasan=lamec")
```

Comment:
@sdogruyol IIRC php as well as ruby use `[]` notation. I'd also point out that there's no way of knowing whether you should construct an `Array` or `String` before you have at least 2 elements. `?arr=1` vs `?arr[]=1` - in the latter case you know the type from the beginning.

 also how would you differentiate